src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimService.java

```diff
@@ -205,13 +209,9 @@ public Claim saveClaim(
         List<String> features
     ) {
         String externalId = claimData.getExternalId().toString();
-        Optional<GeneratePinResponse> pinResponse = Optional.empty();
         User user = userService.getUser(authorisation);
 
-        if (!claimData.isClaimantRepresented()) {
-            pinResponse = Optional.of(userService.generatePin(claimData.getDefendant().getName(), authorisation));
-        }
-
+        Optional<GeneratePinResponse> pinResponse = getPinResponse(claimData, authorisation);
```

Comment:
Should we move pin generation inside try block after this statement? `caseRepository.getClaimByExternalId(externalId, user).ifPresent...`